pkg/annotations_test.go

```diff
@@ -44,3 +46,178 @@ func TestGetSchedule(t *testing.T) {
 		t.Errorf("expected schedule \"*/1 * * * *\", got %s", result)
 	}
 }
+
+func TestGetCronitorID(t *testing.T) {
+	tests := []struct {
+		name                  string
+		annotationIDInference string
+		annotationCronitorID  string
+		expectedID            string
+	}{
+		// Not sure how we can test this, since the expectedID is from the Kubernetes API
+		// {
+		// 	name:                  "default k8s ID",
+		// 	annotationIDInference: "",
+		// 	annotationCronitorID:  "",
+		// 	expectedID:            "kubernetes uid",
+		// },
+		// {
+		// 	name:                  "manual k8s ID",
+		// 	annotationIDInference: "k8s",
+		// 	annotationCronitorID:  "",
+		// 	expectedID:            "kubernetes uid",
+		// },
+		{
+			name:                  "hashed name as ID",
+			annotationIDInference: "name",
+			annotationCronitorID:  "",
+			expectedID:            "7bec37031fa63007a383ade88997bea5bba68d99",
+		},
+		{
+			name:                  "specific cronitor id",
+			annotationIDInference: "",
+			annotationCronitorID:  "1234",
+			expectedID:            "1234",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			annotations := fmt.Sprintf(`"%s": "%s"`, "k8s.cronitor.io/id-inference", tc.annotationIDInference)
+			annotations += fmt.Sprintf(`, "%s": "%s"`, "k8s.cronitor.io/cronitor-id", tc.annotationCronitorID)
+
+			jsonBlob := fmt.Sprintf(`{
```

Comment:
We could probably pull this out into a helper method to make testing these easier, and more standardized.

Reply:
That would be really nice to have!

Reply:
I took the opportunity to refactor things to utilize a new helper func. Looks pretty clean!